pkg/idempotent/redis/redis.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2018 The Moov Authors
+// Use of this source code is governed by an Apache License
+// license that can be found in the LICENSE file.
+
+// redis is a simple inmemory Recorder implementation. This implementation
+// is intended for simple usecases (local dev) and not production workloads.
+package redis
+
+import (
+	"context"
+
+	redis "github.com/gomodule/redigo/redis"
+)
+
+var (
+	// Used for Redis value
+	defaultAddress = "localhost:6379"
+	defaultTimeout = 86400 //The default timeout should be 24 hours.
+	defaultValue   = struct{}{}
+)
+
+func New() *Redis {
+	return &Redis{}
+}
+
+type Redis struct {
+}
+
+func (r *Redis) SeenBefore(key string) (bool, context.Context) {
+	var ctx context.Context
```

Comment:
I don't see why this Context needs to escape `SeenBefore`. Let's create a context using `WithTimeout(context.TODO(), 25*time.Millisecond)`.

Reply:
```
$ go doc context withtimeout
func WithTimeout(parent Context, timeout time.Duration) (Context, CancelFunc)
    WithTimeout returns WithDeadline(parent, time.Now().Add(timeout)).

    Canceling this context releases resources associated with it, so code should
    call cancel as soon as the operations running in this Context complete:

    func slowOperationWithTimeout(ctx context.Context) (Result, error) {
    	ctx, cancel := context.WithTimeout(ctx, 100*time.Millisecond)
    	defer cancel()  // releases resources if slowOperation completes before timeout elapses
    	return slowOperation(ctx)
    }
```